tests/src/test/java/org/eclipse/hono/tests/registry/TenantManagementIT.java

```diff
@@ -455,6 +455,8 @@ public void testGetTenantSucceeds(final VertxTestContext context)  {
             .compose(ar -> getHelper().registry.getTenant(tenantId))
             .onComplete(context.succeeding(httpResponse -> {
                 final JsonObject json = httpResponse.bodyAsJsonObject();
+                // ignore messaging type extension as it is set dynamically by generic test logic
+                json.getJsonObject(TenantConstants.FIELD_EXT).remove(TenantConstants.FIELD_EXT_MESSAGING_TYPE);
```

Comment:
why do we need to remove the messaging type? In fact, we should actually assert that it has a particular value, shouldn't we?

Reply:
This test focuses on tenant specific operations and verifies that the tenant is equal to the JSON which is used as a fixture in the test cases.

The fixture tenant JSON is enriched "under the hood" by _DeviceRegistryHttpClient_. Hence the returned JSON can never be equal the fixture JSON. 

The only other option I see would be to add the the corresponding _messaging_type_ field to _ext_ in _buildTenantPayload_  (which is then overwritten with the same value by _DeviceRegistryHttpClient_). As this involves duplicating code I decided to remove it from the returned JSON.

(The verification of the _messaging_type_ field is done implicitly as all Kafka tests would fail if it wasn't set properly)

Reply:
The Tenant service implementation is always invoked via AMQP and it is expected to return the configured *messaging_type* extension field, isn't it? So FMPOV we should make sure (i.e. assert) that it does.

Reply:
> The fixture tenant JSON is enriched "under the hood" by DeviceRegistryHttpClient. Hence the returned JSON can never be equal the fixture JSON.

IMHO this actually makes it hard(er) to understand the integration tests ...

Reply:
> The Tenant service implementation is always invoked via AMQP and it is expected to return the configured messaging_type extension field, isn't it? So FMPOV we should make sure (i.e. assert) that it does.

I am not sure if I got you right. `TenantManagementIT` calls `IntegrationTestSupport.registry` (via `getHelper().registry`) which is of type `DeviceRegistryHttpClient`. Also the JavaDoc of states it is making calls to the Tenant management's HTTP endpoint.

Do I miss something obvious? Or are you talking about the protocol adapters? If so shouldn't we add the tests to `TenantAmqpIT` then?

Reply:
> IMHO this actually makes it hard(er) to understand the integration tests ...

Yes. It was a trade-off in my opinion. I looked into making this more explicit, but then many test cases which create a new tenant need to be adapted to add the corresponding field to the tenant's `ext` field which would add a lot of boilerplate.

The idea was that `IntegrationTestSupport` takes care of setting up the messaging system specific stuff, so that test cases can be focused on the concrete scenario and code does not need to be duplicated.

Reply:
> Do I miss something obvious? Or are you talking about the protocol adapters? If so shouldn't we add the tests to TenantAmqpIT then?

I'd say we should verify the returned result for both the Registry Management API's Tenant endpoint and the Tenant API.

Reply:
Done!